src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -122,28 +87,61 @@ public DependencyPropertyDetails FindPropertyDetails(DependencyProperty property
 
 		private DependencyPropertyDetails? TryGetPropertyDetails(DependencyProperty property, bool forceCreate)
 		{
-			EnsureEntriesInitialized();
+			if (forceCreate)
+			{
+				var bucketIndex = property.UniqueId >> 4;
+				var bucketRemainder = property.UniqueId & 15;
```

Comment:
What is the 15 magic number? BucketSize - 1?

Reply:
Since BucketSize(= 16) is a power of 2, you can use a bitmask to compute UniqueId % 16

Reply:
Divisions and Modulo are still expensive on modern hardware.

Reply:
Let's explain that in the comments